engine/access/ingestion/tx_error_messages/tx_error_messages_requester.go

```diff
@@ -0,0 +1,75 @@
+package tx_error_messages
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/sethvargo/go-retry"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// TransactionErrorMessagesRequesterConfig contains the retry settings for the tx error messages fetch.
+type TransactionErrorMessagesRequesterConfig struct {
+	// the initial delay used in the exponential backoff for failed tx error messages download
+	// retries.
+	RetryDelay time.Duration
+	// the max delay used in the exponential backoff for failed tx error messages download.
+	MaxRetryDelay time.Duration
+}
+
+// TransactionErrorMessagesRequester is a thin wrapper over a TxErrorMessagesCore that manages the retrieval
+// of transaction error messages with retry and backoff configurations for a specific execution result.
+type TransactionErrorMessagesRequester struct {
+	core            *TxErrorMessagesCore
+	config          *TransactionErrorMessagesRequesterConfig
+	executionResult *flow.ExecutionResult
+}
+
+func NewTransactionErrorMessagesRequester(
+	core *TxErrorMessagesCore,
+	config *TransactionErrorMessagesRequesterConfig,
+	executionResult *flow.ExecutionResult,
+) *TransactionErrorMessagesRequester {
+	return &TransactionErrorMessagesRequester{
+		core:            core,
+		config:          config,
+		executionResult: executionResult,
+	}
+}
+
+// RequestTransactionErrorMessages fetches transaction error messages for the specific
+// execution result this requester was configured with.
+func (r *TransactionErrorMessagesRequester) RequestTransactionErrorMessages(
+	ctx context.Context,
+) error {
+	backoff := retry.NewExponential(r.config.RetryDelay)
+	backoff = retry.WithCappedDuration(r.config.MaxRetryDelay, backoff)
+	backoff = retry.WithJitterPercent(15, backoff)
+
+	blockID := r.executionResult.BlockID
+	resultID := r.executionResult.ID()
+
+	attempt := 0
+	return retry.Do(ctx, backoff, func(context.Context) error {
+		if attempt > 0 {
+			r.core.log.Debug().
+				Str("block_id", blockID.String()).
+				Str("result_id", resultID.String()).
+				Uint64("attempt", uint64(attempt)).
+				Msgf("retrying download")
+		}
+		attempt++
+
+		err := r.core.FetchTransactionResultErrorMessagesByResultID(ctx, blockID, resultID)
+		if errors.Is(err, storage.ErrNotFound) || status.Code(err) == codes.Canceled {
```

Comment:
This check should be reversed. when handling errors we must check for the specific sentinel errors we expect, and fail for all other error types.
I think the expected sentinels are:
* `rpc.ErrNoExecutionReceiptsFoundOnENs` -> no ENs matched for this result
* any grpc status code except `codes.Canceled`

`storage.ErrNotFound` is not retriable since it means that there were no `ExecutionReceipts` in storage for the provided result, which is not possible if the result was used to start a pipeline.